watertap/examples/flowsheets/case_studies/wastewater_resource_recovery/metab/metab.py

```diff
@@ -0,0 +1,627 @@
+
+    # metab_methane
+    m.fs.metab_methane.load_parameters_from_database(use_default_removal=True)
+
+
+def initialize_system(m):
+    seq = SequentialDecomposition()
+    seq.options.tear_set = []
+    seq.options.iterLim = 1
+    seq.run(m, lambda u: u.initialize())
+
+
+def solve(blk, solver=None, tee=False, check_termination=True):
+    if solver is None:
+        solver = get_solver()
+    results = solver.solve(blk, tee=tee)
+    if check_termination:
+        assert_optimal_termination(results)
+    return results
+
+
+def display_results(m):
+    unit_list = ["feed", "metab_hydrogen", "metab_methane"]
+    for u in unit_list:
+        m.fs.component(u).report()
+
+
+def add_costing(m):
+    source_file = os.path.join(
+        os.path.dirname(os.path.abspath(__file__)),
+        "metab_global_costing.yaml",
+    )
+    m.fs.costing = ZeroOrderCosting(default={"case_study_definition": source_file})
+    # typing aid
+    costing_kwargs = {"default": {"flowsheet_costing_block": m.fs.costing}}
+    m.fs.metab_hydrogen.costing = UnitModelCostingBlock(**costing_kwargs)
+    m.fs.metab_methane.costing = UnitModelCostingBlock(**costing_kwargs)
+
+    m.fs.costing.cost_process()
+    m.fs.costing.add_electricity_intensity(m.fs.product_H2O.properties[0].flow_vol)
+
+    # m.fs.costing.add_LCOW(m.fs.product_H2O.properties[0].flow_vol)
+
+    # other levelized costs
+    m.fs.costing.annual_water_production = Expression(
+        expr=m.fs.costing.utilization_factor
+        * pyunits.convert(
+            m.fs.product_H2O.properties[0].flow_vol,
+            to_units=pyunits.m**3 / m.fs.costing.base_period,
+        )
+    )
+    m.fs.costing.annual_cod_removal = Expression(
+        expr=(
+            m.fs.costing.utilization_factor
+            * pyunits.convert(
+                m.fs.feed.outlet.flow_mass_comp[0, "cod"]
+                - m.fs.product_H2O.inlet.flow_mass_comp[0, "cod"],
+                to_units=pyunits.kg / m.fs.costing.base_period,
+            )
+        )
+    )
+    m.fs.costing.annual_hydrogen_production = Expression(
+        expr=(
+            m.fs.costing.utilization_factor
+            * pyunits.convert(
+                m.fs.metab_hydrogen.byproduct.flow_mass_comp[0, "hydrogen"],
+                to_units=pyunits.kg / m.fs.costing.base_period,
+            )
+        )
+    )
+    m.fs.costing.annual_methane_production = Expression(
+        expr=(
+            m.fs.costing.utilization_factor
+            * pyunits.convert(
+                m.fs.metab_methane.byproduct.flow_mass_comp[0, "methane"],
+                to_units=pyunits.kg / m.fs.costing.base_period,
+            )
+        )
+    )
+    m.fs.costing.total_annualized_cost = Expression(
+        expr=(
+            m.fs.costing.total_capital_cost * m.fs.costing.capital_recovery_factor
+            + m.fs.costing.total_operating_cost
+        )
+    )  # TODO: check utilization factor here
```

Comment:
Does this TODO need to be addressed right now? Or is this meant to be a note for later?

Reply:
I think this is outdated.